### DISCUSSION TITLE
Infinite recursion in storage with built-in template

### DISCUSSION THREAD

**Discussion by probert94** [supabase/supabase#22977] [Questions] [answered]
I have a bucket "user" with a folder for each user.
I want every user to be able to access only files inside his own folder.
For this I use the built-in template `Give users access to only their own top level folder named as uid`.
I then enabled `SELECT` and `INSERT` and created the policies without modifying anything.
The resulting statements look like this:
**Select**
```CREATE POLICY "Give users access to own folder 2558r_0" ON storage.objects FOR SELECT TO public USING (bucket_id = 'user' AND (select auth.uid()::text) = (storage.foldername(name))[1]);```

**INSERT**
```CREATE POLICY "Give users access to own folder 2558r_1" ON storage.objects FOR INSERT TO public WITH CHECK (bucket_id = 'user' AND (select auth.uid()::text) = (storage.foldername(name))[1]);```

However this results in the following error when inserting images using supabase-js:
```insert into "objects" ("bucket_id", "metadata", "name", "owner", "owner_id", "version") values ($1, DEFAULT, $2, $3, $4, $5) - infinite recursion detected in policy for relation "objects"```

As soon as I remove the policy for `SELECT`, the `INSERT` works fine but of course I cannot read the files anymore.
Is there anything I am missing?

**Comment by GaryAustin1** [comment, marked as answer]
Do you have any other policies on storage?
I don't see why those two would cause recursion.  

Tested and seem to work fine.

**Reply by probert94** [reply]
You are right, it was a policy for another bucket.
For some reason it only was a problem when I also added a "SELECT" policy for this bucket, so I did not think that it could be related to that other policy.
Thank you
